clash-lib/src/Clash/Netlist/BlackBox/Types.hs

```diff
@@ -83,6 +83,9 @@ data Element = C   !Text         -- ^ Constant
              -- ^ Generated input hole, first argument is the scoping level
              | N   !Int          -- ^ Name hole
              | L   !Int          -- ^ Literal hole
+             | LC  !Int
+             -- ^ Assert that argument is a literal, but treat it like a normal
```

Comment:
Where is this asserted?

Reply:
Ah, I guessed as much but I was thrown off by the docs:

```
-- | Determine if the number of normal/literal/function inputs of a blackbox
-- context at least matches the number of argument that is expected by the
-- template.
```

Maybe it's a good idea to add a small comment? Also: maybe add a `TODO` to get a more explicit error out instead of a general "can't match template".